lib/sql.js

```diff
@@ -889,10 +889,26 @@ SQLConnector.prototype._constructUpdateQuery = function(model, where, fields) {
  * @param {Function} cb The callback function
  */
 SQLConnector.prototype.update = function(model, where, data, options, cb) {
+  var self = this;
   var stmt = this.buildUpdate(model, where, data, options);
-  this._executeAlteringQuery(model, stmt.sql, stmt.params, options, cb || NOOP);
+  this._executeAlteringQuery(model, stmt.sql, stmt.params, options, function(err, info) {
+    if (err) return cb(err);
+    if (where.id && info.count === 0) {
+      return cb(errorIdNotFoundForUpdate(where.id));
+    } else {
+      return cb(null, info) || NOOP;
```

Comment:
@qpresley Actually, since `forceId:true` uses `updateAttributes` and not `update`, I created this pr to solve this issue: https://github.com/strongloop/loopback-connector/pull/100 and `where.id` is not used as a validation connection. I think I will use the other pr instead of this.